### PR TITLE
Checkbox polish

### PR DESCRIPTION
**Pull request by jacob314**
Fyi @terrylucas 

Cleaning up the NotifierCheckbox so it is more convenient to reuse in the inspector page.
This class now gives us a really terse checkbox fully linked to a ValueNotifier. I recommend we use it for all our UI that needs to link checkboxes to ValueNotifier objects in our controllers.

### CODE REVIEW

File: packages/devtools_app/lib/src/ui/utils.dart [hunk 1]
```diff
@@ -7,44 +7,71 @@
...
-/// ValueNotifier associated with a Checkbox widget.
-class CheckboxValueNotifier extends ValueNotifier<bool> {
-  CheckboxValueNotifier(bool value) : super(value);
+class _NotifierCheckboxState extends State<NotifierCheckbox>
+    with AutoDisposeMixin {
+  bool currentValue;
+
+  @override
+  void initState() {
+    // TODO: implement initState
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/ui/utils.dart:41]: remove todo

File: packages/devtools_app/lib/src/memory/memory_filter.dart [hunk 2]
```diff
@@ -139,24 +139,6 @@ class SnapshotFilterState extends State<SnapshotFilterDialog>
     controller = widget.controller;
 
     cancel();
-
-    // Detect and handle checkboxes state changing.
-    addAutoDisposeListener(controller.filterPrivateClassesListenable, () {
-      setState(() {
-        privateClasses.notifier.value = controller.filterPrivateClasses.value;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/memory/memory_filter.dart:146]: are there no other parts of the UI that depend on the value of these notifiers?

File: packages/devtools_app/lib/src/ui/utils.dart (hunk 1 above)
Comment (jacob314) [reply packages/devtools_app/lib/src/ui/utils.dart:41]: done

File: packages/devtools_app/lib/src/memory/memory_filter.dart (hunk 2 above)
Comment (jacob314) [reply packages/devtools_app/lib/src/memory/memory_filter.dart:146]: cleaned up the UI a little further so it is clear that this isn't needed.

**Comment by kenzieschmoll** [comment]
lgtm_army

**Pull request merged**